### PR TITLE
add git branching model

### PR DESCRIPTION
**Pull request by peel**
A brief overview to be a quick reminder what's what in gitflow an links to more elaborate resources 
Source: https://snowplow.slack.com/archives/CKG7E50HX/p1581004086209800

Part of https://github.com/snowplow/snowplow/pull/4283 effort

### CODE REVIEW

File: git-branching.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+- `feature/....` - short-lived branches introducing complete features; spun-off develop and to integrated into develop
+- `hotfix/...` - short-lived branches spun off master to fix the current production version
+- `release/...` -  short-lived, spun off develop to kickstart release process; integrated into master
+
+For further details see resources [resources](#resources)
+
+## Series branches
+
+Additionally, for long-term support we use dedicated `series/x.x` branches that can be spun off already released versions when major or minor has already advanced.
+For example `beam-enrich` version `1.0.0` has been released but due to critical error uncovered in `0.4.0` some changes have to be introduced to the version along with additional backports. This can be done easily by spawning a dedicated `series/0.4` branch and allowing `master` without to advance further without rewrites while maintaining flexibility on further changes to `0.4.x`-series.
```
Comment (lukeindykiewicz) [inline git-branching.md]: extra word, `without to advance further without rewrites`

**Review by lukeindykiewicz — APPROVED** [review]
lgtm!

**Review by benjben — APPROVED** [review]

**Review by chuwy — APPROVED** [review]
LGTM!

Could you also cross-link it with Style Guide and mention that we use it for `beam-enrich` now and going to use for `scala-common-enrich` as an experiment.

**Comment by peel** [comment]
@chuwy I'd rather limit number of reasons for change in that file as docs are not going to age well. Let's put that info in #8 that is referenced in first sentence (I made it more visible), so we don't have to change it over time for that reason.

**Pull request merged**